### DISCUSSION TITLE
Trying to draw an InfiniteLine on mouse-click, but won't display

### DISCUSSION THREAD

**Discussion by ShivaSmith** [highfestiva/finplot#328] [Q&A] [answered]
Hi,

I know I'm processing the mouse click just fine (by using set_time_inspector()), because I can see a print statement execute.

Right after the print(), I'm trying to draw a vertical line on top of the candle I clicked on, using example code from complicated.py.  So I'm trying this:
```python
ax.price_line = pg.InfiniteLine(angle=90, movable=False, pen=fplt._makepen(fplt.draw_line_color))
pen=fplt._makepen(fplt.draw_line_color, style='.'))
ax.price_line.setPos(x)
ax.price_line.pen.setColor(pg.mkColor(fplt.foreground))
ax.addItem(ax.price_line, ignoreBounds=True)
```
But I can't ever see a vertical line be displayed...  I also tried calling fplt.refresh() at the end, which made no difference.  I've also tried various colors, but to no effect.  I'm using the x value that automatically gets passed in via set_time_inspector().

Am I missing something basic with how to get a newly drawn item to display?

Thanks  😄 

p.s.

This application is a bit unusual because I'm using:

```python
app = QApplication([])
win = QMainWindow()
add_menus(win)

fplt.display_timezone = None
fplt.refresh()
# fplt.show()
win.show()
app.exec()
```
Note how I need to use win.show()... can't use fplt.show() when using windows this way.

p.p.s.

My ax is created as:
```python
ax, ax2 = fplt.create_plot('S&P 500 MACD', rows=2, maximize=False)
```

This ax gets added, via:
```python
layout.addWidget(ax.vb.win)
```
in add_menus() --
```python
def add_menus(win):
    main_widget = QWidget()
    layout = QGridLayout()
    main_widget.setLayout(layout)
    layout.addWidget(ax.vb.win)
    win.setCentralWidget(main_widget)
    win.resize(1000, 800)
    win.setWindowTitle('Market Master')

    menu_bar = QMenuBar()
    file_menu = menu_bar.addMenu('File')
    
    menu_bar.show()
    win.setMenuBar(menu_bar)
```
Maybe the way I'm handing my QApplication and QMainWindow means I need to do some other step to get items that are newly added to my ax to display?  It seems like it must be something like that...

**Comment by highfestiva** [comment, marked as answer]
You have to transform your x-coordinate, which is epoch-millisecond time, to a zero-based index. You can use this cumbersome method:

```python
def place_line(x, y):
    pen=fplt._makepen(fplt.draw_line_color, style='.')
    ax.price_line = pg.InfiniteLine(angle=90, movable=False, pen=pen)
    x, = fplt._pdtime2index(ax, pd.Series([x])) # convert timestamp to x-index
    ax.price_line.setPos((x,0))
    ax.addItem(ax.price_line, ignoreBounds=True)

fplt.set_time_inspector(place_line, ax=ax, when='click')
```

**Reply by ShivaSmith** [reply]
Thanks so much!  Yes that was it. 😄 

I had done this conversion:
```python
current_time = pd.to_datetime(x, unit='ns')
```
but didn't realize I needed to use yet a different index 😉 .

Also I had just realized that this method also works:
```python
line = fplt.add_line((current_time, 0), (current_time, 500), color='#9900ff', interactive=True)
```
So now I know 2 methods for lines lol...

Finplot is totally awesome!   
------------------------------
It is so useful!... (perfect for me ;)... super happy that you make it available -- and also how
quickly you answered my question -- very helpful!!  It's allowing me to make the financial charting / drawing / calculation app I always wanted to make :)... having the power of PyQt (and pyqtgraph) in the background is unmatched... embedding my PlotItem inside a QMainWindow let me easily add a QMenuBar up above the plot area, etc... the sky is the limit.  😁

**Reply by highfestiva** [reply]
Thanks for the thanks, and good luck with your trading!
